Subscriber/PaymentMethodSubscriber.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace MollieShopware\Subscriber;
+
+use Enlight\Event\SubscriberInterface;
+use Enlight_Event_EventArgs;
+use Enlight_View;
+use Exception;
+use Mollie\Api\Resources\Method;
+use MollieShopware\Components\Config;
+use MollieShopware\Components\Helpers\MollieShopSwitcher;
+use MollieShopware\Components\Installer\PaymentMethods\PaymentMethodsInstaller;
+use MollieShopware\Components\Payment\Provider\ActivePaymentMethodsProvider;
+use MollieShopware\MollieShopware;
+use Psr\Log\LoggerInterface;
+
+class PaymentMethodSubscriber implements SubscriberInterface
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    /**
+     * @var MollieShopSwitcher
+     */
+    private $mollieShopSwitcher;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param MollieShopSwitcher $mollieShopSwitcher
+     */
+    public function __construct(LoggerInterface $logger, MollieShopSwitcher $mollieShopSwitcher)
+    {
+        $this->logger = $logger;
+        $this->mollieShopSwitcher = $mollieShopSwitcher;
+    }
+
+    /**
+     * @return string[]
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            'Enlight_Controller_Action_PostDispatchSecure_Frontend_Checkout' => 'onFrontendCheckoutPostDispatch',
+        ];
+    }
+
+    /**
+     * @param Enlight_Event_EventArgs $args
+     * @throws Exception
+     */
+    public function onFrontendCheckoutPostDispatch(Enlight_Event_EventArgs $args)
+    {
+        $shop = Shopware()->Shop();
+        $actionName = $args->getRequest()->getActionName();
+        $config = $shop ? $this->mollieShopSwitcher->getConfig($shop->getId()) : null;
+
+        if (!$config || !$config->useMolliePaymentMethodLimits()) {
+            return;
+        }
+
+        if ($actionName !== 'shippingPayment' && $actionName !== 'confirm') {
+            return;
+        }
+
+        /** @var Enlight_View $view */
+        $view = $args->getSubject()->View();
+        $basket = isset($view->sBasket) && is_array($view->sBasket) ? $view->sBasket : [];
+        $value = isset($basket['AmountNumeric']) ? $basket['AmountNumeric'] : '';
+        $currency = isset($basket['sCurrencyName']) ? $basket['sCurrencyName'] : '';
+
+        if ($value === '' || $currency === '') {
+            return;
+        }
+
+        # get all active and available payment methods for a certain amount from Mollie
+        $availableMethods = $this->getActivePaymentMethodsProvider($config)->getActivePaymentMethodsFromMollie(
+            [
```

Comment:
can we get rid off that loose usage here, one could simple screw up a string param in here without recognizing..
can you learn our payment provider to "getByCartAmount"...then we always have 1 single source of truth for our use cases
also for the number_format, please use the NumberFormatter class from the mollie requests, i think that also exits, thank you

Reply:
sry for being confusing on this :D
i meant that the "free array" with configs shouldnt be passed in from here.
so that would then be done inside a getByCartAmount function.
otherwise one could use that getActivePaymentMethods in ANY way, which should not happen (at the moment at least) :)